### PR TITLE
Adding more tests for callback gradients

### PR DESCRIPTION
**Pull request by erick-xanadu**
Just adding more tests

### CODE REVIEW

File: frontend/test/pytest/test_callback.py [hunk 1]
```diff
@@ -1176,7 +1176,7 @@ def fun_bwd_callback(cot) -> jnp.array([1.0, 1.0]):
     def fun_bwd(_res, cot):
         return fun_bwd_callback(cot)
 
-    @qml.qjit(keep_intermediate=True)
+    @qml.qjit
```
Comment (dime10) [inline frontend/test/pytest/test_callback.py:1179]: Thanks, wondered why a new folder popped up when running tests.

**Review by dime10 — APPROVED** [review]
Looks good!

File: frontend/test/pytest/test_callback.py (hunk 1 above)
Comment (erick-xanadu) [reply frontend/test/pytest/test_callback.py:1179]: :sweat_smile:
